refs #371: Added the whole contributions guidelines section (#372)
carenalgas/popochiu: docs/src/contributing-to-popochiu/contributing-documentation/conventions.md

- When referring to numbered markers in screenshots, follow the format `(_<marker_number>_)` (e.g., `(_1_)`, `(_2_)`, `(_32_)`).
- Always link **Godot classes and nodes** to the [official documentation](https://docs.godotengine.org/). For example: [Sprite2D](https://docs.godotengine.org/en/stable/classes/class_sprite2d.html).
- Write class names, functions, variables, file paths, and file names using inline code blocks (e.g., `path/to/file.gd`, `PopochiuCharacter`, `say()`).
- Node names in a scene should always be written in **bold** (e.g., **WalkableAreas**) and, when appropriate, can also be links - always in bold.
- For template code snippets that contain variable elements, enclose the variable part in `<` and `>` markers (e.g., `Popochiu<CharacterName>`). Use the appropriate naming convention for the context (e.g., PascalCase, snake_case, etc).
- Code examples should always be placed in fenced code blocks, specifying the language for syntax highlighting when possible.
- Monospaced text should be placed in fenced code blocks with the `text` or `plaintext` language specified to neutralize the syntax highlighter.

> - Emphasis (`_`) can be used freely and is generally preferred over quotation marks for titles or proper names (e.g., `_Day of the Tentacle_` rather than "Day of the Tentacle"), though this is not a strict rule.

Reviewer: I like this SO SO MUCH!!! 🤓🤓🤓🤓🤓🤓

Author: :partying_face: :balloon:

```markdown
!!! tip "Make your info stand out!"
    Indent with two tabs and **use formatting** for style galore!

    * Empty lines are supported (indent on the new line!)
    * Lists are there as well
    * _You_ **can** do what you want!
```

becomes

!!! tip "Make your info stand out!"
    Indent with two tabs and **use formatting** for style galore!

    * Empty lines are supported (indent on the new line!)
    * Lists are there as well
    * _You_ **can** do what you want!

### Attributes list

This allows to add HTML attributes like id or classes to parts of your Markdown, without littering it with hard-to-read HTML tags.

We use it for _fanced codeblocks_, to add a color code to them:

`````markdown
```{.gdscript .code-example-bad}
print("I'm a bad boy!")
```

```{.gdscript .code-example-good}
print("I'm a good one!")
```
`````

becomes

```{.gdscript .code-example-bad}
print("I'm a bad boy!")
```

```{.gdscript .code-example-good}
print("I'm a good one!")
```

### FontAwesome support

You can print FontAwesome icons in your text in a markdown-ish format:

```markdown

> I :fa-heart: coffee in large :fa-mug-hot:

Reviewer: ❤️❤️❤️❤️❤️

```

becomes

I :fa-heart: coffee in large :fa-mug-hot:

### Definition lists

> Another way to avoid HTML in source code, this extension allows the creation of `<dl>` tags from simpler markdown:

Reviewer: ```suggestion
Another way to avoid HTML in source code. This extension allows the creation of `<dl>` tags from simpler markdown:
```

```markdown
**This is a term or question**
: This is it's definition or answer.  

> Multiple lines are supported (notice the double space in the line above, to create a newline character)

Reviewer: If I'm wrong, ignore this suggestion.

```suggestion
Multiple lines are supported (notice the double space at the end of the line above to create a newline character)
```

Author: Quite the opposite, it's very good!

    C --> D
```
`````

becomes

```mermaid
graph TD
    A["I can"]
    D["Diagrams"]
    A --> B["paint"]
    A --> C["write"]
    B --> D
    C --> D
```

---

## File naming and page titles

When naming Markdown files and directories, clarity is key. Names should follow a "slug format" that corresponds to the page's title, such as `new-section/my-awesome-tutorial.md`. Always use hyphens, never underscores!

To comply with Markdown formatting and ensure proper integration with MkDocs, the first element of each page **must** be a first-level header (`#`), reflecting the page's title. For example:

```markdown
# My Awesome Tutorial
```

This header will be used by MkDocs as the sidebar menu title for that page, overriding the file name.

In certain cases, the file name may differ from the page title for simplicity, as long as the relationship between the two remains clear. For instance, the file named `qna.md` has the title `Contributions Q&A` as its first header.

---

## Images and assets naming conventions

Images that appear in the documentation pages must be placed in the `docs/src/assets/images` folder, keeping them into a subfolder named as the main category they belong.

> Each file (prefer PNG over JPG when possible), must be named accordin to this format: `subsection_name-sorting_number-brief_description.png`.

Reviewer: ```suggestion
Each file (prefer PNG over JPG when possible), must be named according to this format: `subsection_name-sorting_number-brief_description.png`.
```
